contracts/token/Token.sol

```diff
@@ -119,7 +120,13 @@ error DefaultAllowanceAlreadyDisabled(address _user);
 error DefaultAllowanceAlreadySet(address _target);
 
 
-contract Token is IToken, AgentRoleUpgradeable, TokenStorage, IERC165 {
+contract Token is IToken, AgentRoleUpgradeable, TokenStorage, IERC165, TokenPermit {
+    
+    bytes32 private constant _TYPE_HASH =
```

Comment:
we put all the variables and constants in the TokenStorage contract to separate storage from implementation and make sure we don't create collisions when we make an update. Otherwise if it is not practical to use the TokenStorage contract then you need to assign a specific storage slot to the data